bindings/java/src/org/sleuthkit/datamodel/blackboardutils/ArtifactsHelper.java

```diff
@@ -0,0 +1,180 @@
+	 * Creates an artifact helper for artifacts.
+	 *
+	 * @param caseDb     Sleuthkit case db
+	 * @param moduleName name module using the helper
+	 * @param srcFile    source file
+	 *
+	 */
+	public ArtifactsHelper(SleuthkitCase caseDb, String moduleName, AbstractFile srcFile) {
+		super(caseDb, moduleName, srcFile);
+	}
+
+	/**
+	 * Adds a TSK_GPS_TRACKPOINT artifact.
+	 *
+	 * @param latitude    location latitude, required
+	 * @param longitude   location longitude, required
+	 * @param timeStamp   date/time trackpoint recorded, can be 0 if not
+	 *                    available
+	 * @param name        trackpoint name, can be empty/null if not available
+	 * @param programName name of program that recorded trackpoint , can be
+	 *                    empty or null if not available
+	 *
+	 * @return artifact added
+	 */
+	public BlackboardArtifact addGPSLocation(double latitude, double longitude,
+			long timeStamp, String name, String programName) {
+
+		return addGPSLocation(latitude, longitude, timeStamp, name, programName,
+				Collections.<BlackboardAttribute>emptyList());
+	}
+
+	/**
+	 * Adds a TSK_GPS_TRACKPOINT artifact.
+	 *
+	 * @param latitude            location latitude, required
+	 * @param longitude           location longitude, required
+	 * @param timeStamp           date/time trackpoint recorded, can be 0 if not
+	 *                            available
+	 * @param name                trackpoint name, can be empty/null if not
+	 *                            available
+	 * @param programName         name of program that recorded trackpoint , can
+	 *                            be empty or null if not available
+	 * @param otherAttributesList other attributes, can be empty list of no
+	 *                            additional attributes
+	 *
+	 * @return artifact added
+	 */
+	public BlackboardArtifact addGPSLocation(double latitude, double longitude, long timeStamp, String name, String programName,
+			Collection<BlackboardAttribute> otherAttributesList) {
+
+		BlackboardArtifact gpsTrackpointArtifact = null;
+		try {
+			Collection<BlackboardAttribute> attributes = new ArrayList<>();
+
+			// Create artifact
+			gpsTrackpointArtifact = getAbstractFile().newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_TRACKPOINT);
+
+			// Add basic attributes 
+			attributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_LATITUDE, getModuleName(), latitude));
+			attributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_LONGITUDE, getModuleName(), longitude));
+
+			addAttributeIfNotZero(timeStamp, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME, attributes);
+
+			addAttributeIfNotNull(name, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_NAME, attributes);
+			addAttributeIfNotNull(programName, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PROG_NAME, attributes);
+
+			// add the attributes 
+			gpsTrackpointArtifact.addAttributes(attributes);
+			gpsTrackpointArtifact.addAttributes(otherAttributesList);
```

Comment:
Why make two database transactions and two round trips here? The two collections of attributes can be combined and added in a single addAttributes call. 

This comment applies to multiple locations in this PR.